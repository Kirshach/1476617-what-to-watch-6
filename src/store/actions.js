export const ActionType = {
  SET_GENRE: `films/setGenre`,
};

export const setGenre = (payload) => ({
  type: ActionType.SET_GENRE,
  payload
});
