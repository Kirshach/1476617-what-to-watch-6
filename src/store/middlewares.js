import {history} from '../history';

export const redirectActionType = `app/redirect`;

export const redirectAction = (payload) => ({
  type: redirectActionType,
  payload,
});

export const redirectMiddleware = (_store) => (next) => (action) => {
  if (action.type === redirectActionType) {
    history.push(action.payload);
  }
  return next(action);
};
