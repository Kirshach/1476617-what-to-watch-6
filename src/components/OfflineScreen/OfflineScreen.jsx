import React from 'react';
import {useDispatch} from 'react-redux';

import {setIsOnlineAction} from '../../store/app/state/actions';

import PageFooter from '../PageFooter/PageFooter';
import PageHeader from '../PageHeader/PageHeader';

const OfflineScreen = () => {
  const dispatch = useDispatch();

  const onConnectMeButtonClick = () => {
    if (window.navigator.onLine) {
      dispatch(setIsOnlineAction(true));
    }
  };

  return (
    <div className="user-page">
      <PageHeader className="user-page__head" noUserBlock/>

      <main className="info-screen">
        <h2 className="info-screen__heading">Oops...</h2>
        <p className="info-screen__message">It looks like you&apos;re offline ðŸ‘€</p>
        <p className="info-screen__message">Press here if you think you&apos;ve restored connection:</p>
        <button className="btn info-screen__btn" type="button" onClick={onConnectMeButtonClick}>
          <span>Connect me!</span>
        </button>
      </main>

      <PageFooter />
    </div>
  );
};

OfflineScreen.propTypes = {};

export default OfflineScreen;
