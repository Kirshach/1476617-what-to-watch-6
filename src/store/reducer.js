import {ActionType} from './actions';
import {Genres} from '../const';
import {films} from '../mocks/films';


const initialState = {
  films,
  genre: Genres.allGenres,
  genres: Object.values(Genres),
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return {
        ...state,
        genre: action.payload
      };
    default:
      return state;
  }
};

export {reducer};
