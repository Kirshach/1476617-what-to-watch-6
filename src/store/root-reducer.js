import {combineReducers} from 'redux';

import {appReducer} from './app/reducer';
import {domainReducer} from './domain/reducer';

export const DOMAIN = `domain`;
export const APP = `app`;

const rootReducer = combineReducers({
  [DOMAIN]: domainReducer,
  [APP]: appReducer,
});

export {rootReducer};

export const RootNS = {DOMAIN, APP};
