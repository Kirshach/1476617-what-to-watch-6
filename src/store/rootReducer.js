import {combineReducers} from 'redux';

import {appReducer} from './app/reducer';
import {domainReducer} from './domain/reducer';

const DOMAIN = `domain`;
const APP = `app`;

const rootReducer = combineReducers({
  [DOMAIN]: domainReducer,
  [APP]: appReducer,
});

export {rootReducer};

export const RootNS = {DOMAIN, APP};
