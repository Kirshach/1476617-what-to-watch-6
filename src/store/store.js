import {createStore, applyMiddleware} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import thunk from 'redux-thunk';

import {redirectMiddleware, handleAPIErrorMiddleware} from './middlewares';
import {checkAuthThunk} from '../store/app/auth/thunks';
import {fetchFilmsThunk} from '../store/domain/thunks';
import {rootReducer as reducer} from '../store/rootReducer';

import {api} from '../api/api';

const store = createStore(
    reducer,
    composeWithDevTools(
        applyMiddleware(
            handleAPIErrorMiddleware,
            redirectMiddleware,
            thunk.withExtraArgument(api),
        ),
    ),
);

store.dispatch(checkAuthThunk());
store.dispatch(fetchFilmsThunk());

export {store};
