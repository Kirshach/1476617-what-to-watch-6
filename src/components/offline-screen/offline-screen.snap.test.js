import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router';
import {useDispatch, useSelector} from 'react-redux';

import {store} from '../../mocks/store/store';

import OfflineScreen from './offline-screen';

jest.mock(`react-redux`);

test(`<MyList /> renders correctly`, () => {
  useDispatch.mockReturnValue(() => Promise.resolve());
  useSelector.mockImplementation((selector) => selector(store));

  const {container} = render(
      <MemoryRouter>
        <OfflineScreen />
      </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
