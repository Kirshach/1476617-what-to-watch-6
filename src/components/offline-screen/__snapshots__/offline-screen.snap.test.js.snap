// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyList /> renders correctly 1`] = `
<div>
  <div
    class="user-page"
  >
    <header
      class="page-header user-page__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
    </header>
    <main
      class="info-screen"
    >
      <h2
        class="info-screen__heading"
      >
        Oops...
      </h2>
      <p
        class="info-screen__message"
      >
        It looks like you're offline ðŸ‘€
      </p>
      <p
        class="info-screen__message"
      >
        Press here if you think you've restored connection:
      </p>
      <button
        class="btn info-screen__btn"
        type="button"
      >
        <span>
          Connect me!
        </span>
      </button>
    </main>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
